Shared/Realm.Sync.Shared/Permissions/PermissionChange.cs

```diff
@@ -40,59 +39,114 @@ public class PermissionChange : RealmObject, IPermissionObject
 
         /// <summary>
-        /// Gets or sets read access.
+        /// Gets a value indicating whether the user(s) have read access to the specified Realm(s).
         /// </summary>
         /// <value><c>true</c> or <c>false</c> to request this new value. <c>null</c> to keep current value.</value>
         [MapTo("mayRead")]
-        public bool? MayRead { get; set; }
+        public bool? MayRead { get; private set; }
 
         /// <summary>
-        /// Gets or sets write access.
+        /// Gets a value indicating whether the user(s) have write access to the specified Realm(s).
         /// </summary>
         /// <value><c>true</c> or <c>false</c> to request this new value. <c>null</c> to keep current value.</value>
         [MapTo("mayWrite")]
-        public bool? MayWrite { get; set; }
+        public bool? MayWrite { get; private set; }
 
         /// <summary>
-        /// Gets or sets manage access.
+        /// Gets a value indicating whether the user(s) have manage access to the specified Realm(s).
         /// </summary>
         /// <value><c>true</c> or <c>false</c> to request this new value. <c>null</c> to keep current value.</value>
         [MapTo("mayManage")]
-        public bool? MayManage { get; set; }
+        public bool? MayManage { get; private set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PermissionChange"/> class.
+        /// </summary>
+        /// <param name="userId">The user or users who should be granted these permission changes. Use * to change permissions for all users.</param>
+        /// <param name="realmUrl">The Realm URL whose permissions settings should be changed. Use `*` to change the permissions of all Realms managed by the management Realm's <see cref="User"/>.</param>
+        /// <param name="mayRead">Define read access. <c>true</c> or <c>false</c> to request this new value. <c>null</c> to keep current value.</param>
+        /// <param name="mayWrite">Define write access. <c>true</c> or <c>false</c> to request this new value. <c>null</c> to keep current value.</param>
+        /// <param name="mayManage">Define manage access. <c>true</c> or <c>false</c> to request this new value. <c>null</c> to keep current value.</param>
+        public PermissionChange(string userId, string realmUrl, bool? mayRead = null, bool? mayWrite = null, bool? mayManage = null)
+        {
+            UserId = userId;
+            RealmUrl = realmUrl;
+            MayRead = mayRead;
+            MayWrite = mayWrite;
+            MayManage = mayManage;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PermissionChange"/> class.
+        /// </summary>
+        /// <remarks>
+        /// This constructor is used by Realm to create a new instance from C++. You should use the constructor that accepts parameters instead.
+        /// </remarks>
+        public PermissionChange()
```

Comment:
If you make this internal or even private the woven object helper should still be able to access it as it's a nested class.